File: Examples/CSharp/Text/InsertChineseNumberList.cs
using System;
using System.Drawing;

namespace Aspose.Note.Examples.CSharp.Text
{
    public class InsertChineseNumberList
    {
        public static void Run()
        {
            // ExStart:InsertChineseNumberList
            // ExFor:NumberList
            // ExFor:ParagraphStyle
            // ExFor:Style.FontColor
            // ExFor:Style.FontName
            // ExFor:Style.FontSize
            // ExFor:Page
            // ExFor:Outline
            // ExFor:OutlineElement
            // ExFor:OutlineElement.NumberList
            // ExFor:RichText
            // ExFor:RichText.ParagraphStyle
            // ExFor:RichText.Text
            // ExSummary:Shows how to insert new list with chinese numbering.

            string dataDir = RunExamples.GetDataDir_Text();

            // Initialize OneNote document
            Document doc = new Document();

            // Initialize OneNote page
            Page page = new Page();
            Outline outline = new Outline();

            // Apply text style settings
            ParagraphStyle defaultStyle = new ParagraphStyle { FontColor = Color.Black, FontName = "Arial", FontSize = 10 };

            // Numbers in the same outline are automatically incremented.
            OutlineElement outlineElem1 = new OutlineElement() { NumberList = new NumberList("{0})", NumberFormat.ChineseCounting, "Arial", 10) };
            RichText text1 = new RichText() { Text = "First", ParagraphStyle = defaultStyle };
            outlineElem1.AppendChildLast(text1);
            
            //------------------------
            OutlineElement outlineElem2 = new OutlineElement() { NumberList = new NumberList("{0})", NumberFormat.ChineseCounting, "Arial", 10) };
            RichText text2 = new RichText() { Text = "Second", ParagraphStyle = defaultStyle };
            outlineElem2.AppendChildLast(text2);
            
            //------------------------
            OutlineElement outlineElem3 = new OutlineElement() { NumberList = new NumberList("{0})", NumberFormat.ChineseCounting, "Arial", 10) };
            RichText text3 = new RichText() { Text = "Third", ParagraphStyle = defaultStyle };
            outlineElem3.AppendChildLast(text3);

            //------------------------
            outline.AppendChildLast(outlineElem1);
            outline.AppendChildLast(outlineElem2);
            outline.AppendChildLast(outlineElem3);
            page.AppendChildLast(outline);
            doc.AppendChildLast(page);

            // Save OneNote document
            dataDir = dataDir + "InsertChineseNumberList_out.one"; 
            doc.Save(dataDir);
            
            // ExEnd:InsertChineseNumberList

            Console.WriteLine("\nChinese number list inserted successfully.\nFile saved at " + dataDir);
        }
    }
}

File: Examples/VisualBasic/Text/InsertChineseNumberList.vb
Imports System.Globalization
Imports System.IO
Imports AsposeNote = Aspose.Note
Imports System.Drawing
Imports System.Collections.Generic
Namespace Text
    Public Class InsertChineseNumberList
        Public Shared Sub Run()
            ' ExStart:InsertChineseNumberList
            Dim dataDir As String = RunExamples.GetDataDir_Text()

            ' Initialize OneNote document
            Dim doc As New AsposeNote.Document()
            ' Initialize OneNote page
            Dim page As New AsposeNote.Page(doc)
            Dim outline As New Outline(doc)
            ' Apply text style settings
            Dim defaultStyle As New TextStyle() With {
                 .FontColor = Color.Black,
                 .FontName = "Arial",
                 .FontSize = 10
            }

            ' Numbers in the same outline are automatically incremented.
            Dim outlineElem1 As New OutlineElement(doc) With {
                 .NumberList = New NumberList("{0})", NumberFormat.ChineseCounting, "Arial", 10)
            }
            Dim text1 As New RichText(doc) With {
                 .Text = "First",
                 .DefaultStyle = defaultStyle
            }
            outlineElem1.AppendChild(text1)
            '------------------------
            Dim outlineElem2 As New OutlineElement(doc) With {
                 .NumberList = New NumberList("{0})", NumberFormat.ChineseCounting, "Arial", 10)
            }
            Dim text2 As New RichText(doc) With {
                 .Text = "Second",
                 .DefaultStyle = defaultStyle
            }
            outlineElem2.AppendChild(text2)
            '------------------------
            Dim outlineElem3 As New OutlineElement(doc) With {
                 .NumberList = New NumberList("{0})", NumberFormat.ChineseCounting, "Arial", 10)
            }
            Dim text3 As New RichText(doc) With {
                 .Text = "Third",
                 .DefaultStyle = defaultStyle
            }
            outlineElem3.AppendChild(text3)
            '------------------------
            outline.AppendChild(outlineElem1)
            outline.AppendChild(outlineElem2)
            outline.AppendChild(outlineElem3)
            page.AppendChild(outline)
            doc.AppendChild(page)

            dataDir = dataDir & Convert.ToString("InsertChineseNumberList_out_.one")
            ' Save OneNote document
            doc.Save(dataDir)
            ' ExEnd:InsertChineseNumberList
            Console.WriteLine(Convert.ToString(vbLf & "Chinese number list inserted successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace